activesupport/lib/active_support/cache/redis_cache_store.rb

```diff
@@ -118,7 +118,9 @@ class << self
         def build_redis(redis: nil, url: nil, **redis_options) #:nodoc:
           urls = Array(url)
 
-          if redis.respond_to?(:call)
+          if redis.is_a?(Redis)
+            redis
+          elsif redis.respond_to?(:call)
             redis.call
           elsif redis
             redis
```

Comment:
I left this branch untouched. My rationale is that someone may want to provide an object that is _workalike_ with Redis (ie. implementing `get`, `set`, etc. with a different backend) but that doesn't inherit from Redis. Should this branch remain?

Reply:
Hmm... the triple branch does seem a bit ugly.

If we're going to have to use an `is_a?` check to distinguish them, maybe we'd be better changing the existing condition to `is_a?(Proc)`?

Reply:
Sure, SGTM

Reply:
Pushed